docs/rules/no-thenable.md

```diff
@@ -0,0 +1,63 @@
+# Disallow `then` property
+
+✅ *This rule is part of the [recommended](https://github.com/sindresorhus/eslint-plugin-unicorn#recommended-config) config.*
+
+If an object is defined as "thenable", once it's accidentally used in an await expression, it may cause problems.
+
+If a module have exported `then`, dynamic `import()` may won't works as expected.
```

Comment:
```suggestion
If a module has an export named `then`, dynamic `import()` may not work as expected.
```

This also should describe how it will not work.